ipaplatform/redhat/authconfig.py

```diff
@@ -18,15 +18,102 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import logging
+
 from ipaplatform.paths import paths
 from ipapython import ipautil
 from ipapython.admintool import ScriptError
 import os
 
 FILES_TO_NOT_BACKUP = ['passwd', 'group', 'shadow', 'gshadow']
 
+logger = logging.getLogger(__name__)
+
+
+class RedHatAuthTool(object):
+    auth_tool = None
+
+    @classmethod
+    def getInstance(cls):
+        if cls.auth_tool is None:
+            cls.auth_tool = RedHatAuthSelect()
+        return cls.auth_tool
+
+
+class RedHatAuthToolBase(object):
+
+    def configure(self, sssd, mkhomedir, statestore):
+        raise NotImplementedError(
+            'RedHatAuthToolBase.configure() not implemented'
+        )
+
+    def unconfigure(self, fstore, statestore,
+                    was_sssd_installed,
+                    was_sssd_configured):
+        raise NotImplementedError(
+            'RedHatAuthToolBase.unconfigure() not implemented'
+        )
+
+    def backup(self, path):
+        raise NotImplementedError(
+            'RedHatAuthToolBase.backup() not implemented'
+        )
 
-class RedHatAuthConfig(object):
+    def restore(self, path):
+        raise NotImplementedError(
+            'RedHatAuthToolBase.restore() not implemented'
+        )
+
+
+class RedHatAuthSelect(RedHatAuthToolBase):
+
+    def configure(self, sssd, mkhomedir, statestore):
+        if not sssd:
+            logger.error("no-sssd options is not supported by installer. "
+                         "For client installation without sssd check the "
+                         "output of 'ipa-advise config-fedora-authconfig'")
+            return
+            # raise ScriptError("Non-sssd installation is not supported")
+
+        # Do not raise exception since authselect may fail just
+        # because there is PAM configuration already in place.
+        if mkhomedir:
+            ipautil.run(
+                [
+                    paths.AUTHSELECT,
+                    "select", "sssd",
+                    "with-mkhomedir"
+                ],
+                raiseonerr=False
```

Comment:
Why ``raiseonerr=False``? It's dangerous because it hides bugs and errors.

Reply:
answer to the question is in the comment above. the thing is that authselect is failing if some profile is already activated.
we can use --force option to authselect but then it is silently overwrite configuration. I think it is better to make the tool fail and inform a user about this.
If we raise an exception, an installer will fail with a traceback which is oufull.

Reply:
Does authselect fail with a specific error number? If authselect has reserved a return code for "profile already active", then you could check the return code and only ignore that error number.

Reply:
ok. will do.